broker/src/test/java/io/camunda/zeebe/broker/system/partitions/impl/steps/BackupServiceTransitionStepTest.java

```diff
@@ -0,0 +1,183 @@
+  void setup() {
+    transitionContext.setConcurrencyControl(TEST_CONCURRENCY_CONTROL);
+    transitionContext.setActorSchedulingService(actorSchedulingService);
+    transitionContext.setBackupStore(backupStore);
+    transitionContext.setBrokerCfg(brokerCfg);
+    transitionContext.setRaftPartition(raftPartition);
+
+    lenient().when(brokerCfg.getCluster()).thenReturn(clusterCfg);
+    lenient().when(clusterCfg.getPartitionsCount()).thenReturn(3);
+    lenient()
+        .when(raftPartition.members())
+        .thenReturn(Set.of(MemberId.from("1"), MemberId.from("2")));
+    lenient().when(raftPartition.dataDirectory()).thenReturn(Path.of("/tmp/zeebe").toFile());
+
+    lenient()
+        .when(actorSchedulingService.submitActor(any()))
+        .thenReturn(TEST_CONCURRENCY_CONTROL.completedFuture(null));
+
+    lenient()
+        .when(backupManagerPreviousRole.closeAsync())
+        .thenReturn(TEST_CONCURRENCY_CONTROL.completedFuture(null));
+
+    step = new BackupServiceTransitionStep();
+  }
+
+  @ParameterizedTest
+  @ArgumentsSource(TransitionsThatShouldCloseService.class)
+  void shouldCloseExistingService(final Role currentRole, final Role targetRole) {
+    // given
+    transitionContext.setCurrentRole(currentRole);
+    if (currentRole != null && currentRole != Role.INACTIVE) {
+      transitionContext.setBackupManager(backupManagerPreviousRole);
+      transitionContext.setCheckpointProcessor(recordsProcessorPreviousRole);
+    }
+
+    // when
+    step.prepareTransition(transitionContext, 1, targetRole).join();
+
+    // then
+    assertThat(transitionContext.getBackupManager()).isNull();
+    verify(backupManagerPreviousRole).closeAsync();
+    assertThat(transitionContext.getCheckpointProcessor()).isNull();
+  }
+
+  @ParameterizedTest
+  @ArgumentsSource(TransitionsThatShouldInstallService.class)
+  void shouldReInstallService(final Role currentRole, final Role targetRole) {
+    // given
+    transitionContext.setCurrentRole(currentRole);
+    if (currentRole != null && currentRole != Role.INACTIVE) {
+      transitionContext.setBackupManager(backupManagerPreviousRole);
+      transitionContext.setCheckpointProcessor(recordsProcessorPreviousRole);
+    }
+
+    // when
+    transitionTo(targetRole);
+
+    // then
+    assertThat(transitionContext.getBackupManager())
+        .isNotNull()
+        .isNotEqualTo(backupManagerPreviousRole);
+    assertThat(transitionContext.getCheckpointProcessor())
+        .isNotNull()
+        .isNotEqualTo(recordsProcessorPreviousRole);
+  }
+
+  @ParameterizedTest
+  @ArgumentsSource(TransitionsThatShouldDoNothing.class)
+  void shouldNotReInstallService(final Role currentRole, final Role targetRole) {
+    // given
+    transitionContext.setCurrentRole(currentRole);
+    if (currentRole != null && currentRole != Role.INACTIVE) {
+      transitionContext.setBackupManager(backupManagerPreviousRole);
+      transitionContext.setCheckpointProcessor(recordsProcessorPreviousRole);
+    }
```

Comment:
:upside_down_face: This could be it's own method IMO, `setupForCurrentRole` or something. This would make it a bit clearer that the same rules for setting up the state apply to all tests.